src/test/java/com/HubertKarw/medical_clinic/Service/PatientJpaServiceTest.java

```diff
@@ -0,0 +1,102 @@
+        when(patientJpaRepository.findAll(any(Pageable.class))).thenReturn(page);
+        //when
+        List<Patient> result = patientJpaService.getPatients(PageRequest.of(0, 3));
+        //then
+        assertAll(
+                () -> assertEquals(3, result.size()),
+                () -> assertEquals(123, result.get(0).getId()),
+                () -> assertEquals(124, result.get(1).getId()),
+                () -> assertEquals(125, result.get(2).getId())
+        );
+    }
+
+    @Test
+    void getPatient_patientExists_patientReturned() {
+        //given
+        User user = new User(1l, "123", "321");
+        Patient patient = new Patient(123L, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        when(patientJpaRepository.findByEmail("x1")).thenReturn(Optional.of(patient));
+        //when
+        Patient result = patientJpaService.getPatient("x1");
+        //then
+        assertEquals(patient.getId(), result.getId());
+    }
+
+    @Test
+    void addPatient_validPatient_patientAdded() {
+        //given
+        User user = new User(1l, "123", "321");
+        Patient patient = new Patient(null, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        Patient patientSaved = new Patient(123L, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        when(patientJpaRepository.save(any())).thenReturn(patientSaved);
+        //when
+        Patient result = patientJpaService.addPatient(patient);
+        //then
+        assertEquals(patient.getEmail(), patientSaved.getEmail());
+    }
+
+    @Test
+    void removePatient_patientExists_patientRemoved() {
+        //given
+        User user = new User(1l, "123", "321");
+        Patient patient = new Patient(null, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        Patient patientSaved = new Patient(123L, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        when(patientJpaRepository.findByEmail(any())).thenReturn(Optional.of(patient));
+        //when
+        patientJpaService.removePatient(patient.getEmail());
+        //then
+        assertEquals(1, 1);
+    }
+
+    @Test
+    void modifyPatient_patientExists_patientModified() {
+        User user = new User(1l, "123", "321");
+        Patient patient = new Patient(null, "x1", user, "123", "xyz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        Patient modifiedPatient = new Patient(123L, "x1", user, "123", "xyzz", "xzy", "123", LocalDate.of(1999, 11, 11));
+        when(patientJpaRepository.findByEmail(any())).thenReturn(Optional.of(patient));
+        when(patient.update(any(Patient.class))).thenReturn(modifiedPatient);
+        when(patientJpaRepository.save(any())).thenReturn(modifiedPatient);
+        //when
+        patientJpaService.modifyPatient("x1", modifiedPatient);
+        //then
+        assertNotEquals(patient.getFirstName(), modifiedPatient.getFirstName());
```

Comment:
assertEquals(expected, actual) 
expected czyli to co uwazamy ze powino zostac zwrocone a actual to rezultat wykonania testu czyli w tym przypadku nasz obiekt result